File: Leetcode/167-Two-Sum-II---Input-array-is-sorted/TwoSumII.go
func twoSum(numbers []int, target int) []int {
	if numbers == nil || len(numbers) < 2 {
		return []int{-1, -1}
	}
	left := 0
	length := len(numbers)
	right := length - 1
	for left < right {
		sum := numbers[left] + numbers[right]
		if sum == target {
			return []int{left + 1, right + 1}
		} else if sum > target {
			right--
		} else {
			left++
		}
	}
	return []int{-1, -1}
}

File: Leetcode/167-Two-Sum-II---Input-array-is-sorted/TwoSumII.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        vector<int> res;
        int left=0, right=numbers.size()-1;
        while (left < right) {
            int sum = numbers[left] + numbers[right];
            if (sum == target) {
                res.push_back(left+1);
                res.push_back(right+1);
                break;
            } else if (sum < target) {
                left++;
            } else {
                right--;
            }
        }
        return res;
    }
};


